jcl/src/java.base/share/classes/java/lang/invoke/VarargsCollectorHandle.java

```diff
@@ -99,13 +101,39 @@ static MethodType varargsCollectorType(MethodType nextType, Class<?> arrayType)
 	}
 	
 	public Object invokeWithArguments(Object... args) throws Throwable, WrongMethodTypeException, ClassCastException {
-		if (args.length < 253) {
-			MethodHandle mh = IWAContainer.getMH(args.length);
-			return mh.invokeExact((MethodHandle)this, args);
+		int argsLength = 0;
+		if (args != null) {
+			argsLength = args.length;
+		}
+		if (argsLength < 253) {
+			MethodHandle mh = IWAContainer.getMH(argsLength);
+			return mh.invokeExact((MethodHandle) this, args); 
+		/*[IF Java10]*/
```

Comment:
I think this should be Java11, not 10

Reply:
it seems to have been an enhancement in 10 that we missed https://docs.oracle.com/javase/10/docs/api/java/lang/invoke/MethodHandle.html#invokeWithArguments(java.lang.Object...)